packages/node-firebird-driver/src/lib/impl/attachment.ts

```diff
@@ -194,6 +194,10 @@ export abstract class AbstractAttachment implements Attachment {
 		return statement;
 	}
 
+	isValid() {
+		return !!this.client;
+	}
+
 	private check() {
 		if (!this.client)
```

Comment:
Please rewrite `check` methods to call `isValid` instead of duplicate the same conditions.